### PR TITLE
Split up related field node type in data abstraction layer

### LINKED ISSUE #19929: Split up o2m vs m2o relational types

### PR DESCRIPTION
**Pull request by nickrum**
## Scope

What's changed:

- The first two commits are from directus/directus#19941
- Split up related field node type in data `@directus/data` to account for the fact that m2o and a2o relations do not have any nested modifiers
- Added option to add "object nested" and "json nested" field types in the future

## Potential Risks / Drawbacks

- Even more nesting in the `AbstractQuery` type

## Review Questions

- I'm not too sure about the `meta` field name for the relational info. I'm open to suggestions!

---

Fixes directus/durus#11

### CODE REVIEW

**Comment by jaads** [comment]
we can actually remove the term `node` out of `field-nodes.ts` since everything else is basically a node too and no other file has this in its name

File: packages/data/src/types/abstract-query/fields/nested.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export interface AbstractQueryFieldNodeNestedMany {
+	type: 'nested-many';
+
+	/* From the related collection the user can pick primitives, apply a function or add another nested node   */
+	fields: AbstractQueryFieldNode[];
+	alias?: string;
+
+	/** For man it always possible to add modifiers to the foreign collection to adjust the results. */
```
Comment (jaads) [inline packages/data/src/types/abstract-query/fields/nested.ts:25]: tiny grammar fix
```suggestion
	/** For many it's always possible to add modifiers to the foreign collection to adjust the results. */
```

**Review by jaads — APPROVED** [review]
nice 👍 for me it's no risk or drawback that it got more nested, I like this split up.

an alternative to `meta` could be `relation` 🤔 but both would be fine :)

**Review by rijkvanzanten — APPROVED** [review]
Makes a lot of sense to me 👍🏻

**Pull request merged**